### PR TITLE
MySQL APB example

### PR DESCRIPTION
**Pull request by codificat**
An example APB to deploy MySQL using the rhscl images.

Still WIP but submitting for feedback.

Known pending tasks:

- [x] confirm the *plan* approach to persistent storage. Revert to bool param if not yet ready
- [x] consider using a template for the DC instead of 2 separate dev/prod DCs
- [x] adjust `/etc/apb/bind-creds` so that it works under a restricted SCC
- [x] update `com.redhat.apb.spec` with the latest changes
- [ ] squash and rebase

### CODE REVIEW

**Comment by jcpowermac** [comment]
LGTM

File: rhscl-mysql-apb/Makefile [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+####### Test 1: Provision the "dev" plan (no storage)
+	oc project ${PROJ_RANDOM}
+	oc run ${IMAGE_NAME} --image=${REGISTRY}:5000/${PROJ_RANDOM}/${IMAGE_NAME} --restart=OnFailure -- provision -e namespace=${APB_APP} -e plan=dev
+# With --restart=OnFailure it creates a Job instead of a DC, so the following won't work:
+#	oc rollout status -w dc/${IMAGE_NAME}
+#	oc describe dc/${IMAGE_NAME}
+#	oc logs -f dc/${IMAGE_NAME}
+# FIXME: it's creating a pod, not a Job. Also fix the 'delete pod's below
+# FIXME: wait for the Job to complete with something better than a sleep/logs
```
Comment (tchughesiv) [inline rhscl-mysql-apb/Makefile]: +1 ... I'm going to research this a bit.

Comment (tchughesiv) [reply rhscl-mysql-apb/Makefile]: @codificat here's what i just did for hello-world-apb... moved to pod run and added logic to watch pod for completion...
https://github.com/fusor/apb-examples/commit/d84b7666cece64f481b17d6178203809697b7f21

Comment (codificat) [reply rhscl-mysql-apb/Makefile]: Thanks @tchughesiv - I'm pushing now an alternative approach using `--attach=true` in the `oc run` invocation. This should take care of showing the logs of the APB, waiting for its termination and exit with a matching exit code

**Comment by dymurray** [comment]
@codificat I merged in two other APBs that probably conflicted with your Makefile and Travis changes. Please rebase and I will start testing.

**Comment by codificat** [comment]
Thanks @dymurray - Rebased and updated. Travis tests pass (which shows that they are not complete because they don't report a mismatch between the current apb-base and this image in bind-creds location...)

The main question I have is: this is using a plan to determine storage reqs. Are you ok with that or should we wait a bit more? (the commit history contains the previous implementation using a bool param instead of a plan, can revert to that)

I'll squash before merging

**Comment by dymurray** [comment]
After talking with @codificat and @eriknelson our plan is to leave this PR OPEN until we implement plans in the Broker. @eriknelson will use this APB to test out the broker's implementation and this PR will be updated to match the final design.

**Comment by eriknelson** [comment]
I'm marking this as a WIP since the broker does not support plans yet and this APB may require some minor changes as we build things out on the broker side. I'm hoping to have plan support in the broker sometime in the next week so we'll have a broker that can actually deploy this.

Thank you @codificat, super helpful.

**Comment by codificat** [comment]
@dymurray I believe this is getting closer to be able to merge. I updated it to match recent changes in plan mgmt.

Known pending things:

- travis tests are currently failing (since credentials passing switched to exec-based) because there's no broker that gets the bind credentials from the APB. I think this would fail with any other bindable APB that had similar tests in place... I'll probably update Makefiles on a separate PR if that's ok?

- Rebase and squash all these commits before merging (I've been keeping history here to be able to look at progress/look back)

At this point though I'd like to get your thoughts about the content of the APB itself.

Thanks!

**Comment by dymurray** [comment]
@codificat sorry for the delay. I tested your APB and it worked great on my end. I was able to deploy both plans and verified mysql came up. Please squash and rebase and I am happy to merge. I am fine with updating the makefile after.

**Pull request merged**